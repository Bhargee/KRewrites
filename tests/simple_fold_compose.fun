fold (fun a -> fun b -> a + b) map (fun c -> c * 2)
