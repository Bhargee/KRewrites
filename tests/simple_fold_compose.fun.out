Search results:

Solution 1:
<T>
    <k>
        (((fold) (fun a -> (fun b -> (a + b)))) ((map) (fun c -> (c * 2)))
          )
    </k>
    <store>
        .Map
    </store>
    <env>
        .Map
    </env>
</T>

Solution 2:
<T>
    <k>
        ((fold) ((fun a -> (fun b -> (a + b))) -| (fun c -> (c * 2))))
          
    </k>
    <store>
        .Map
    </store>
    <env>
        .Map
    </env>
</T>
